src/eventSchemas/models/schemaCodeGenUtils.ts

```diff
@@ -0,0 +1,103 @@
+    private readonly AWS_EVENTS_PREFIX = `${this.AWS}.` // . suffix because of 1p event registry schema format
+
+    public buildSchemaPackageName(schemaName: string): string {
+        const builder = new CodeGenPackageBuilder()
+        builder.append(this.SCHEMA_PACKAGE_PREFIX)
+        this.buildPackageName(builder, schemaName)
+
+        return builder.build()
+    }
+
+    private buildPackageName(builder: CodeGenPackageBuilder, schemaName: string): void {
+        // do not modify the order of conditional checks
+        if (this.isAwsPartnerEvent(schemaName)) {
+            this.buildPartnerEventPackageName(builder, schemaName)
+        } else if (this.isAwsEvent(schemaName)) {
+            this.buildAwsEventPackageName(builder, schemaName)
+        } else {
+            this.buildCustomPackageName(builder, schemaName)
+        }
+    }
+
+    private isAwsPartnerEvent(schemaName: string): boolean {
+        return schemaName.startsWith(this.AWS_PARTNER_PREFIX)
+    }
+
+    private buildPartnerEventPackageName(builder: CodeGenPackageBuilder, schemaName: string): void {
+        const partnerSchemaString = schemaName.substring(this.AWS_PARTNER_PREFIX.length)
+
+        builder
+            .append(this.AWS)
+            .append(this.PARTNER)
+            .append(partnerSchemaString)
+    }
+
+    private isAwsEvent(name: string): boolean {
+        return name.startsWith(this.AWS_EVENTS_PREFIX)
+    }
+
+    private buildAwsEventPackageName(builder: CodeGenPackageBuilder, schemaName: string): void {
+        const awsEventSchemaParts = schemaName.split('.')
+        for (const part of awsEventSchemaParts) {
+            builder.append(part)
+        }
+    }
+
+    private buildCustomPackageName(builder: CodeGenPackageBuilder, schemaName: string): void {
+        builder.append(schemaName)
+    }
+}
+
+class CodeGenPackageBuilder {
+    private builder = ''
+    private readonly formatter = IdentifierFormatter.getInstance()
+
+    public build(): string {
+        return this.builder
+    }
+
+    public append(segment: String): CodeGenPackageBuilder {
+        if (this.builder.length > 0) {
+            this.builder = this.builder.concat(this.formatter.PACKAGE_SEPARATOR)
+        }
+        this.builder = this.builder.concat(this.formatter.toValidIdentifier(segment.toLowerCase()))
+
+        return this
+    }
+}
+
+export class IdentifierFormatter {
+    private static INSTANCE: IdentifierFormatter | undefined
+    public readonly PACKAGE_SEPARATOR = '.'
+    private readonly POTENTIAL_PACKAGE_SEPARATOR = '@'
+    private readonly NOT_VALID_IDENTIFIER_CHARACTER = `[^a-zA-Z0-9_${this.POTENTIAL_PACKAGE_SEPARATOR}]`
+    private readonly NOT_VALID_IDENTIFIER_REGEX = new RegExp(this.NOT_VALID_IDENTIFIER_CHARACTER, 'g')
+    private readonly POTENTIAL_PACKAGE_SEPARATOR_REGEX = new RegExp(this.POTENTIAL_PACKAGE_SEPARATOR, 'g')
+    private readonly UNDERSCORE = '_'
```

Comment:
Nit: Do we need this many variables? Will these values ever change? 

At the very least, I feel like `NOT_VALID_IDENTIFIER_CHARACTER` can be rolled directly into the regex without losing anything. Also not super huge on `UNDERSCORE`, considering it's not an abstract concept (it's just an underscore). Lastly, `PACKAGE_SEPARATOR` is not used outside this class and does not need to be public.

Reply:
Well I feel like defining these variables with descriptive names makes it easier to understand the class purpose. The fields could change in the future, we previously had `. `as package separator for instance.  The long-term plan is to replace these calculations and retrieve the results from api calls.  
`PACKAGE_SEPARATOR` is actually used in CodeGenPackageBuilder class. I will remove the
`NOT_VALID_IDENTIFIER_CHARACTER`.